### DISCUSSION TITLE
Packages and scripts for installing signal-cli as systemd daemon

### DISCUSSION THREAD

**Discussion by bublath** [AsamK/signal-cli#753] [Show and tell]
Since the question came up in another discussion, I like to share my solution for installing signal-cli as service.
My repository for signal-cli integration into the home automation system FHEM also includes current 0.9.0 binary libraries as well as a script that will create .deb packages to conveniently install signal-cli.
Check out: https://github.com/bublath/FHEM-Signalbot

If you run the script _build-signal-cli-packages.sh_ it will create a couple of debian packages (Debian 10, Ubuntu 18.04, Ubuntu 20.04, Raspberry Buster) in your home directory, that you can directly install with "apt" or "dpkg".
You will also need signal-cli checked out and built first or use the binary distribution.
Important is that you edit the script to point $SIGNALSOURCE to the directory containing the built _bin_ and _lib_ .
You can also use the command line options to just build one combination, e.g.
_./build-signal-cli-packages.sh amd64 glibc2.28_ 
for Debian 10
The script will automatically replace the binary .so in the corresponding .jar files to match your architecture.

If you prefer to do it yourself, check out the "package" directory which contains the underlying configuration files.

Hope this helps.

**Comment by pbiering** [comment]
I've did the job for RPM based OS like Fedora and Enterprise Linux:

- GitHub (sources): https://github.com/pbiering/signal-cli-rpm
- COPR (ready-to-use RPMs): https://copr.fedorainfracloud.org/coprs/pbiering/InternetServerExtensions/package/signal-cli/

If enough voters will be found I can try to request the package for Fedora upstream and EPEL.
